### ISSUE TITLE
picture decode error,why?

### ISSUE THREAD

**Issue by zhouXiaoR** [rs/SDWebImage#1773]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by mythodeia** [comment]
please use the latest version of the library. also share the decode error you mention.
or you can try the image url that causes the issue in the sample project and check if the error occurs there as well.

**Comment by zhouXiaoR** [comment]
When loading the images, sometimes the images appear black figure, white figure, or the image is not completely, Xcode will show the warning (JPEGDecompressSurface : Picture decode failed:).Whether for thread safety problem ?
![48292121ea8dab737d769ad3da701a6a](https://cloud.githubusercontent.com/assets/19583003/22017303/be62ea72-dce4-11e6-8bc8-e488e22b61b5.jpg)
![00314745887051cd5cd6447c0dc21fad](https://cloud.githubusercontent.com/assets/19583003/22017306/be8aa8dc-dce4-11e6-9007-b6a8981e148e.jpg)

**Comment by mythodeia** [comment]
as i have mentioned above please try the images you sent me in the demo project using the latest version of the library. if you still face issues please let us know

**Comment by huangqiaobo** [comment]
I see the log at 3.8.2
JPEGDecompressSurface : Picture decode failed: e00002d1

**Comment by zhouXiaoR** [comment]
@huangqiaobo 我也在控制台看到了这样的警告错误（Picture decode failed:），所以我也在怀疑，是我们服务端数据传输的错误，还是SDWebImage的Bug,或者，能不能有一个系统的或者自定义的方法来判断某个解码出来的图片是否有误，这样便于再次请求。

**Comment by zhouXiaoR** [comment]
@mythodeia ，thanks your reply, we are  looking for the cause of "Picture decode failed".Looking forward to your reply again!Thanks.

**Comment by mythodeia** [comment]
can you use the latest [release](https://github.com/rs/SDWebImage/releases/tag/4.0.0-beta2) and try again? if the problem still occurs please share the url of some of the images that cause the issue

**Comment by IOSGeekerOfChina** [comment]
@zhouXiaoR 你是不是使用了 SDWebImageOptions 中设置了 SDWebImageProgressiveDownload 如果是 可以去掉就不报错了 应该是有些图片类型不允许 通过SDWebImageProgressiveDownload 这种渐进式的方式下载。
